source/preprocessor.tex

```diff
@@ -45,10 +45,36 @@
     \terminal{\#} conditionally-supported-directive
 \end{bnf}
 
+\begin{bnf}
+\nontermdef{embed-parameter}\br
+    embed-standard-parameter\br
+    embed-prefixed-parameter
+\end{bnf}
+
+\begin{bnf}
+\nontermdef{embed-parameter-seq}\br
+    embed-parameter \opt{embed-parameter-seq}
+\end{bnf}
```

Comment:
This weirdly switches between forward references and no forward references.  Please move embed-parameter-seq to immediately before embed-parameter.